### PR TITLE
Check functor arguments when computing placement of doc comments

### PR DESCRIPTION
**Pull request by Julow**
Partial fix to https://github.com/ocaml-ppx/ocamlformat/issues/1010

Functor arguments are not checked to compute the `force_before` value.

In this example, the comment should be before:

```ocaml
module Make (Config : sig
  val blah : string

  (* this could be a really long signature *)
end) : S
(** Doc comment still goes after *)
```

### CODE REVIEW

File: test/passing/doc_comments-after.ml.ref [hunk 1]
```diff
@@ -107,6 +107,30 @@ module Comment_placement : sig
...
+  (** This one goes after *)
+
+  module Index2
+      (Paramater_module : BAR_LONG_MODULE_TYPE_NAME)
+      (Foo : BAR)
+      (Foo : BAR)
+      (Foo : BAR)
+      (Foo : BAR)
+      (Foo : BAR) : sig end
+  (** This one _still_ goes after *)
```
Comment (Julow) [inline test/passing/doc_comments-after.ml.ref]: Should we do something to this ?
The `*_is_simple` functions could easily be modified to make an estimation of the size of the line, as any corner cases will be not "simple".

Comment (gpetiot) [reply test/passing/doc_comments-after.ml.ref]: It would be nice to do something, maybe just considering that `Pmty_functor (_, Some t1, t2)` is not simple, even if `t1` is simple.

Comment (Julow) [reply test/passing/doc_comments-after.ml.ref]: What about `Pmty_functor (_, None, t2)` ? I changed so any functor is not simple. I have no diffs with `test_branch`, even on [Irmin](https://github.com/mirage/irmin/).

Comment (gpetiot) [reply test/passing/doc_comments-after.ml.ref]: This case is something like `module Foooo () : t2` so we can keep checking if t2 is simple like it's currently done (and add some tests).

Comment (Julow) [reply test/passing/doc_comments-after.ml.ref]: Modifying the _is_simple function won't work because it's not called on the whole module expr/type. But this case can be handled easily in `fmt_module`. I wonder if it's worth adding a "ctx" argument to it to be able to implement better the `single_line` logic ?

I made generative functors simple in `fmt_module`.

Comment (gpetiot) [reply test/passing/doc_comments-after.ml.ref]: > I wonder if it's worth adding a "ctx" argument to it to be able to implement better the `single_line` logic ?

I'm not sure we should over-engineer this function, the output looks good as it is now.
